src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    private int $lastChecked = 0;
+
+    public function __construct(private ContainerInterface $container, private int $checkTiming)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $connection = $this->container->get('database_connection');
```

Comment:
This is a good first step. If it works great, I think something should be contributed to the bundle to tag all connection services so that they can be more easily injected.